include/caffe/util/gpu_memory.hpp

```diff
@@ -0,0 +1,121 @@
+#ifndef CAFFE_UTIL_GPU_MEMORY_HPP_
+#define CAFFE_UTIL_GPU_MEMORY_HPP_
+
+#include <vector>
+#include "caffe/common.hpp"
+
+namespace caffe {
+
+class gpu_memory {
+ public:
+  enum PoolMode {
+    NoPool,     // Straight CUDA malllc/free. May be very expensive
+    CubPool,     // CUB caching allocator
+#ifdef CPU_ONLY
+    DefaultPool = NoPool
+#else
+    DefaultPool = CubPool     // CUB pool is able to use unified memory properly
+#endif
+  };
+
+  static const char* getPoolName();
```

Comment:
As earlier, accessors are `underscore_case` and do not start with `get`. (Cheap methods may also be `underscore_case`. Most methods are `CamelCase`. No methods are `lowercaseCamel`.)